nameko_amqp_retry/backoff.py

```diff
@@ -130,13 +131,15 @@ def republish(self, backoff_exc, message, target_queue):
 
         amqp_uri = self.container.config[AMQP_URI_CONFIG_KEY]
 
-        # force redeclaration; the publisher will skip declaration if
+        # force redeclaration;
+        # In kombu versions prior to 4.3.0 the publisher will skip declaration if
         # the entity has previously been declared by the same connection
         # (see https://github.com/celery/kombu/pull/884)
         conn = Connection(amqp_uri)
-        maybe_declare(
-            queue, conn.channel(), retry=True, **DEFAULT_RETRY_POLICY
-        )
+        if parse_version(kombu_version) < parse_version('4.3.0'):
```

Comment:
moving the parsing to the module level would slightly optimise it as it does not need to be done for each message
```
need_maybe_declare = parse_version(kombu_version) < parse_version('4.3.0')
```

Reply:
true 👍